docs/data/data-grid/components/overview.md

```diff
@@ -0,0 +1,94 @@
+# Data Grid - Components
```

Comment:
@samuelsycamore keen to get your thoughts on the structure/contents of the new "Components" section. In summary:
- We are introducing some new composable components to the Data Grid to help users to extend the data grid.
- The composition and customization APIs are slightly different to what Data Grid and Material UI users are used to, which is why I have included this "Overview" page (not sure if that's the best name) to explain some of the new concepts.

Reply:
+1 for naming this page Usage since we already have Overview pages elsewhere that serve a different purpose.